From b9c0beac8f021f774f39df54e1f96fd87c2660f0 Mon Sep 17 00:00:00 2001
From: Varun Reddy Yeturu <varunreddy.yeturu@codeaurora.org>
Date: Wed, 27 Sep 2017 18:27:10 -0700
Subject: qcacld-3.0: Check for valid vdev ID in wma_nlo_match_evt_handler

Check if the firmware is passing a valid vdev ID or not in the
NLO match event and return error if vdev is invalid

Change-Id: I83f957ae084e17c20f607eb3862a131f3b311d23
CRs-Fixed: 2114776
---
 core/wma/src/wma_scan_roam.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/wma/src/wma_scan_roam.c b/core/wma/src/wma_scan_roam.c
index c366a00..8b1cd7a 100644
--- a/core/wma/src/wma_scan_roam.c
+++ b/core/wma/src/wma_scan_roam.c
@@ -3809,6 +3809,12 @@ int wma_nlo_match_evt_handler(void *handle, uint8_t *event,
 	nlo_event = param_buf->fixed_param;
 	WMA_LOGD("PNO match event received for vdev %d", nlo_event->vdev_id);
 
+	if (nlo_event->vdev_id >= wma->max_bssid) {
+		WMA_LOGE("Invalid vdev id in the NLO event %d",
+				nlo_event->vdev_id);
+		return -EINVAL;
+	}
+
 	node = &wma->interfaces[nlo_event->vdev_id];
 	if (node)
 		node->nlo_match_evt_received = true;
-- 
cgit v1.1

